src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexFindOptimizations.cs

```diff
@@ -10,28 +12,55 @@ namespace System.Text.RegularExpressions
     /// <summary>Contains state and provides operations related to finding the next location a match could possibly begin.</summary>
     internal sealed class RegexFindOptimizations
     {
-        /// <summary>True if the input should be processed right-to-left rather than left-to-right.</summary>
-        private readonly bool _rightToLeft;
         /// <summary>Lookup table used for optimizing ASCII when doing set queries.</summary>
         private readonly uint[]?[]? _asciiLookups;
 
-        public RegexFindOptimizations(RegexNode root, RegexOptions options)
+        public static RegexFindOptimizations Create(RegexNode root, RegexOptions options)
```

Comment:
Why was this changed to a factory vs. constructor chaining? Doesn't really matter I guess other than some private setters had to be added.

Reply:
To more easily handle the bag of results that comes back from multiple calls. We first create find optimizations for the whole pattern, and then based on what comes back, we may instead want to create the optimizations for just a subexpression.